frontend/src/pages/search/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import { GetServerSideProps } from 'next'
+import Link from 'next/link'
+import { SWRConfig } from 'swr'
+import { unstable_serialize } from 'swr/infinite'
+
+import Page from '@components/Page'
+import Project from '@models/Project'
+import {
+  makeSWRKeyGetter,
+  searchFetcher,
+  useLazySearch,
+} from '@hooks/useLazySearch'
+import ProjectCardGrid from '@components/ProjectCardGrid'
+import { useSearchQuery } from '@contexts/SearchContext'
+
+import styles from './index.module.scss'
+
+export const getServerSideProps: GetServerSideProps = async ({
+  query: queryParams,
+}) => {
+  const query = (queryParams.q as string) || ''
+
+  const searchParams = { query }
+
+  const hits = await searchFetcher(searchParams)
+
+  return {
+    props: {
+      swrFallback: {
+        [unstable_serialize(makeSWRKeyGetter(searchParams))]: [hits],
+      },
+      initialQuery: query,
+    },
+  }
+}
+
+interface SearchPageProps {
+  initialQuery: string
+  swrFallback: Record<string, Project[][]>
```

Comment:
```suggestion
  swrFallback: Record<string, Array<Array<Project>>>
```

Reply:
Wait, why is this one an array of array but in `[user]/index.tsx` it's just array?

https://github.com/kitspace/kitspace-v2/pull/504/files/53f8def9ded873b0b88fcde04567ecde09175b3e#diff-14d5e273a8f5b377e25c3f2871911a7390c78dd991031efbbf018ef4fe59d294R44

Reply:
I didn't type the `props` returned in the `getServerSideProps` return value so it didn't catch it. Fixed it 0085daf.